### PR TITLE
[KEYCLOAK-9809] Support private_key_jwt authentication for external IdP

### PR DESCRIPTION
**Pull request by madgaet**

### CODE REVIEW

**Comment by madgaet** [comment]
By looking at the following pull request (https://github.com/keycloak/keycloak/pull/6268)
To be more close to that solution, I could propose another alternative for this ticket.

**Review by thomasdarimont — COMMENTED** [review]
Thanks for your PR - looks good so far and I think this can work in conjunction with #6268.
Small note here: your formatting seems a bit off? I'd suggest to align your settings with the keycloak formatting standards to avoid formatting only changes.

File: services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java [hunk 1]
```diff
@@ -442,13 +454,47 @@ public Response authResponse(@QueryParam(AbstractOAuth2IdentityProvider.OAUTH2_P
...
+        	jwt.issuer(getConfig().getClientId());
+        	jwt.subject(getConfig().getClientId());
+        	jwt.audience(getConfig().getTokenUrl());
+        	jwt.issuedNow();
+        	jwt.expiration(Time.currentTime() + realm.getAccessCodeLifespan());
+        	return jwt;
+        }
+        
+        protected SignatureSignerContext getSignatureContext() {
+			if (getConfig().getClientSecret() != null && !getConfig().getClientSecret().trim().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (madgaet) [reply services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java]: Thanks for the review, I'll have a look at what happened with the formatting.

I'll fix that.

**Comment by madgaet** [comment]
I will create a new pull request

**Pull request closed without merging**